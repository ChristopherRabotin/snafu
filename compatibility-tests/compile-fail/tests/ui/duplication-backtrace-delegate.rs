use snafu::{prelude::*, Backtrace};

#[derive(Debug, Snafu)]
enum EnumError {
    AVariant {
        // First backtrace, legitimate
        backtrace: Backtrace,

        // Second backtrace, this time a delegate, can't have both
        #[snafu(backtrace)]
        source: String,
    },
}

fn main() {}
