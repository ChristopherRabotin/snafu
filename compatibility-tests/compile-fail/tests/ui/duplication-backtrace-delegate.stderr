error: Cannot have `backtrace` field and `backtrace` attribute on a source field in the same variant
  --> tests/ui/duplication-backtrace-delegate.rs:10:9
   |
10 | /         #[snafu(backtrace)]
11 | |         source: String,
   | |______________________^

error: Cannot have `backtrace` field and `backtrace` attribute on a source field in the same variant
 --> tests/ui/duplication-backtrace-delegate.rs:7:9
  |
7 |         backtrace: Backtrace,
  |         ^^^^^^^^^^^^^^^^^^^^
